player/loadfile.c

```diff
@@ -282,13 +293,20 @@ static void print_stream(struct MPContext *mpctx, struct track *t)
     if (t->type == STREAM_VIDEO) {
         if (s && s->codec->disp_w)
             APPEND(b, " %dx%d", s->codec->disp_w, s->codec->disp_h);
-        if (s && s->codec->fps)
-            APPEND(b, " %.3ffps", s->codec->fps);
+        if (s && s->codec->fps && !t->image) {
+            char *fps = mp_format_double(NULL, s->codec->fps, 4, false, false, true);
+            APPEND(b, " %s fps", fps);
+            talloc_free(fps);
+        }
     } else if (t->type == STREAM_AUDIO) {
         if (s && s->codec->channels.num)
-            APPEND(b, " %dch", s->codec->channels.num);
-        if (s && s->codec->samplerate)
-            APPEND(b, " %dHz", s->codec->samplerate);
+            APPEND(b, " %d ch", s->codec->channels.num);
+        if (s && s->codec->samplerate) {
+            char *samplerate = mp_format_double(NULL, s->codec->samplerate / 1000.0,
+                                                4, false, false, true);
+            APPEND(b, " %s kHz", samplerate);
+            talloc_free(samplerate);
+        }
```

Comment:
this is now inconsistent with literally the next line I see after tracks:
`AO: [pipewire] 48000Hz stereo 2ch floatp`

I propose to revert these

Reply:
I propose to fix `AO` output, which is currently the outlier, every other places uses spaces.

Reply:
no, it's mixed:
```
ilters/f_swresample.c
165:    MP_VERBOSE(p, "%dHz %s %s -> %dHz %s %s\n",

stream/dvb_tune.c
464:        MP_VERBOSE(priv, "tuning %s to %d Hz, bandwidth: %d\n",

player/audio.c
295:    snprintf(buf, buf_sz, "%dHz %s %dch %s", rate,

audio/aframe.c
441:    snprintf(buf, buf_size, "%dHz %s %dch %s", fmt->av_frame->sample_rate,

audio/out/ao_wasapi_utils.c
359:    MP_VERBOSE(ao, "Accepted as %s %s @ %dhz -> %s (%s)\n",

audio/out/ao_sdl.c
132:    MP_VERBOSE(ao, "requested format: %d Hz, %d channels, %x, "
145:    MP_VERBOSE(ao, "obtained format: %d Hz, %d channels, %x, "

audio/out/ao_pcm.c
150:    MP_INFO(ao, "File: %s (%s)\nPCM: Samplerate: %d Hz Channels: %d Format: %s\n",

video/out/vulkan/context_display.c
177:            mp_msg(log, msgl_info, "      Mode %02d: %dx%d (%02d.%03d Hz)\n", k,

audio/out/ao.c
212:    MP_VERBOSE(ao, "requested format: %d Hz, %s channels, %s\n",

audio/out/ao_oss.c
221:    MP_VERBOSE(ao, "Using %d Hz samplerate.\n", samplerate);
```

```
audio/aframe.c
441:    snprintf(buf, buf_size, "%dHz %s %dch %s", fmt->av_frame->sample_rate,

player/loadfile.c
303:            APPEND(b, " %d ch", s->codec->channels.num);

audio/out/ao_sdl.c
132:    MP_VERBOSE(ao, "requested format: %d Hz, %d channels, %x, "
145:    MP_VERBOSE(ao, "obtained format: %d Hz, %d channels, %x, "

audio/out/ao_oss.c
191:            MP_ERR(ao, "Failed to set audio device to %d channels.\n",
204:            MP_ERR(ao, "Failed to set audio device to %d channels.\n",
213:        MP_VERBOSE(ao, "Using %d channels (requested: %d).\n",

player/audio.c
295:    snprintf(buf, buf_sz, "%dHz %s %dch %s", rate,

audio/out/ao_alsa.c
808:        MP_ERR(ao, "Asked for %d channels, got %d - fallback to %s.\n", req,
```

`Hz` is debatable but for channels IMO it's either full length with space (`%d channels`) or short form without (`%dch`)

Reply:
Thinking about it I don't particularity like `ch` suffix. `%d channels` is nice, but at this point, we could switch to channel layout which is `2.0`, `5.1` and so on. More concise and doesn't need suffix, because it is obvious what does it mean and how much channels it is.

But I don't feel strongly either way. I personally would go for `2.0` instead `2ch`, but I'm fine with whatever you decide.